lib/conversion-form-helper.ts

```diff
@@ -1,25 +1,35 @@
 import {
   volumeConversions,
   weightConversions,
-  VolumeUnit,
-  WeightUnit,
-  TemperatureUnit,
+  Unit, VolumeUnit,
 } from "./constants";
 
+function isVolumeUnit(unit: Unit): unit is VolumeUnit {
+  return Object.values(VolumeUnit).includes(unit as VolumeUnit);
+}
+
 export function convertVolume(
   value: number,
-  fromUnit: VolumeUnit | WeightUnit | TemperatureUnit,
-  toUnit: VolumeUnit | WeightUnit | TemperatureUnit
+  fromUnit: Unit,
+  toUnit: Unit,
 ): number {
+  if (!isVolumeUnit(fromUnit)) {
```

Comment:
Not the biggest fan of this `Unit` typing but it's a fair price to pay to have a clean form. I do want to make sure these functions break when they are being called incorrectly though.

Question: Right now everything works without issues but I wonder if we need to add an error boundary to the app to catch these exceptions?

Reply:
I am personally fine with leaving this as is.

What's the alternative to having `Unit` here?

Reply:
I don't know if there is a good alternative unless typescript has features I'm not aware of or don't remember (like those fancy pick or union types I never fully know how to use)